src/controllers/productController.js

```diff
@@ -1,14 +1,17 @@
-const errorHandler = require("../utils/errorHandler")
-const validateCreateProductInput = require("../validation/createProduct")
-const productService = require('../services/productService');
+const errorHandler = require('../utils/errorHandler')
+const validateCreateProductInput = require('../validation/createProduct')
+const productService = require('../services/productService')
 
-const createProduct = async (req , res) => {
-    const {success , error , value} = validateCreateProductInput(req.body).validate()
```

Comment:
we can later modify this validate function to return only error , value for simplicity
then we can check if there is a validation error if the error object is not defined

Reply:
do u mean to put it in try-catch block and catch an error if there exist ? xD